### PR TITLE
feat: added can and can_any methods to dpp::permission class

### PR DESCRIPTION
**Pull request by Commandserver**
- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.
- [x] I agree to the terms of the [DCO (Developer Certificate of Origin)]((https://dpp.dev/coding-standards.html))

The new methods do also return true, if the permission has the administrator flag set. Equal to the dpp::role methods

### CODE REVIEW

File: include/dpp/permissions.h [hunk 1]
```diff
@@ -127,6 +127,44 @@ class DPP_EXPORT permission {
...
+	 *
+	 * @return bool True if it has **all** the given permissions or dpp::p_administrator
+	 */
+	template <typename... T>
+	constexpr bool can(T... values) const noexcept {
+		return has(values...) || (value & p_administrator);
+	}
+
+	/**
+	 * @brief Check for certain permissions, taking into account administrator privileges. It uses the Bitwise AND operator
```
Comment (Mishura4) [inline include/dpp/permissions.h:150]: It uses the Bitwise OR operator**

Comment (Commandserver) [reply include/dpp/permissions.h:150]: huh. it uses the has method which uses the AND operator

Comment (Commandserver) [reply include/dpp/permissions.h:150]: well i could remove the "It uses the Bitwise AND operator" entirely as this is irrelevant there

Comment (Mishura4) [reply include/dpp/permissions.h:150]: Oh nvm yeah it uses AND after OR i got confused oop

**Review by braindigitalis — APPROVED** [review]

**Review by Mishura4 — APPROVED** [review]

**Pull request merged**